File: Frontend/legacy/react-pangul/src/application/services/apis/adminDbController.js
export class AdminDbController {
  constructor(fetch, logger) {
    this.fetch = fetch;
    this.logger = logger;
  }

  async getMigrationStatus() {
    try {
      const response = await this.fetch.post('/fetch/adminDb/migrationStatus');
      if (response.ok) {
        const data = await response.json();
        if (data.success) {
          return data.data;
        }
      }
      this.logger.error('Failed', response);
    } catch (error) {
      this.logger.error(error);
    }
    throw new Error('API request failed');
  }
}

File: Backend/Pangul.Backend.Web/Controllers/Admin/AdminDb/AdminDbController.cs
using System;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using NLog;
using Pangul.Backend.Web.Configuration.Authentication.Policy;

namespace Pangul.Backend.Web.Controllers.Admin.AdminDb
{
  [Route("api/[controller]/[action]")]
  [Authorize(Policy = PangulAdminDatabase.PolicyName)]
  public class AdminDbController : Controller
  {
    private AdminDbService _serivce;
    private readonly Logger _logger;

    public AdminDbController()
    {
      _serivce = new AdminDbService();
      _logger = LogManager.GetCurrentClassLogger();
    }

    [HttpPost]
    public async Task<JsonResult> MigrationStatus()
    {
      try
      {
        var result = await _serivce.GetMigrationStatus();
        return Json(new {Success = true, Data = result});
      }
      catch (Exception error)
      {
        _logger.Error(error);
        return Json(new {Success = false});
      }
    }
  }
}